packages/server/src/fhir/operations/codesystemlookup.ts

```diff
@@ -0,0 +1,88 @@
+import { Coding } from '@medplum/fhirtypes';
+import { getOperationDefinition } from './definitions';
+import { parseInputParameters, sendOutputParameters } from './utils/parameters';
+import { Request, Response } from 'express';
+import { TypedValue, allOk, badRequest, notFound } from '@medplum/core';
+import { Column, Condition, SelectQuery } from '../sql';
+import { sendOutcome } from '../outcomes';
+import { getClient } from '../../database';
+
+const operation = getOperationDefinition('CodeSystem', 'lookup');
+
+type CodeSystemLookupParameters = {
+  code: string;
+  system: string;
+  coding: Coding;
+  property: string[];
+};
+
+export async function codeSystemLookupHandler(req: Request, res: Response): Promise<void> {
+  const params = parseInputParameters<CodeSystemLookupParameters>(operation, req);
+
+  let coding: Coding;
+  if (params.coding) {
+    coding = params.coding;
+  } else if (params.system && params.code) {
+    coding = { system: params.system, code: params.code };
+  } else {
+    sendOutcome(res, badRequest('No coding specified'));
+    return;
+  }
+
+  const lookup = new SelectQuery('Coding');
+  const codeSystemTable = lookup.getNextJoinAlias();
+  lookup.innerJoin(
+    'CodeSystem',
+    codeSystemTable,
+    new Condition(new Column('Coding', 'system'), '=', new Column(codeSystemTable, 'id'))
+  );
+  const propertyTable = lookup.getNextJoinAlias();
+  lookup.leftJoin(
+    'Coding_Property',
+    propertyTable,
+    new Condition(new Column(propertyTable, 'coding'), '=', new Column('Coding', 'id'))
+  );
+  const csPropTable = lookup.getNextJoinAlias();
+  lookup.innerJoin(
+    'CodeSystem_Property',
+    csPropTable,
+    new Condition(new Column(propertyTable, 'property'), '=', new Column(csPropTable, 'id'))
+  );
+  lookup
+    .column(new Column(codeSystemTable, 'title'))
+    .column(new Column('Coding', 'display'))
+    .column(new Column(csPropTable, 'code'))
+    .column(new Column(csPropTable, 'type'))
+    .column(new Column(csPropTable, 'description'))
+    .column(new Column(propertyTable, 'value'))
+    .where(new Column(codeSystemTable, 'url'), '=', coding.system)
+    .where(new Column('Coding', 'code'), '=', coding.code);
```

Comment:
How do project permissions work with this query?

Reply:
That's actually originally how I had structured this, but I changed it and the permissions got lost in the shuffle; I'm open to reverting back to this approach.  I also tried to add a simpler check to verify that the CodeSystem is in the user's project: I could see an argument that execution permission on the operation is different from reading the resource, and not being able to read the resource itself shouldn't necessarily block someone from leveraging the terminology service features.  Thoughts either way?